### PR TITLE
[Feature](function) support random int from specific range

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed changes

Issue Number: close #xxx

support:
```sql
mysql> select rand(-20, -10);
+------------------+
| random(-20, -10) |
+------------------+
|              -13 |
+------------------+
1 row in set (0.10 sec)
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/Random.java [hunk 1]
```diff
@@ -58,6 +59,17 @@ public Random(Expression arg) {
         Preconditions.checkState(arg instanceof Literal, "The param of rand function must be literal");
     }
 
+    /**
+     * constructor with 1 argument.
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/Random.java:63]: comment is error

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/Random.java [hunk 2]
```diff
@@ -80,13 +92,15 @@ public boolean foldable() {
...
-            return this;
-        } else {
+        if (children.isEmpty()) {
+            return new Random();
+        } else if (children.size() == 1) {
             return new Random(children.get(0));
+        } else if (children.size() == 2) {
+            return new Random(children.get(0), children.get(1));
         }
+        Preconditions.checkArgument(false, "random function only accept 0-2 arguments");
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/Random.java:102]: throw AnalysisException, do not use check false

**Comment by zclllyybb** [comment]
run buildall

**Review by Gabriel39 — DISMISSED** [review]

**Comment by zclllyybb** [comment]
run pipelinex_p0 10

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Review by Mryange — APPROVED** [review]

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**